src/PrestaShopBundle/DependencyInjection/Compiler/LoadModulesAsBundlesPass.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\Config\Resource\FileExistenceResource;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Config\FileLocator;
+
+/**
+ * Load services stored in modules.
+ */
+class LoadModulesAsBundlesPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $this->registerServicesFromModules($container);
+        $this->registerTemplatesFromModules($container);
+    }
+
+    /**
+     * Load all services registered in every module.
+     *
+     * @param ContainerBuilder $container
+     * @return void
+     */
+    private function registerServicesFromModules(ContainerBuilder $container)
+    {
+        $modulesPaths = $this->getModulesPaths();
+
+        foreach ($modulesPaths as $modulePath) {
+            if (file_exists($modulePath.'/config/services.yml')) {
+                $loader = new YamlFileLoader($container, new FileLocator($modulePath.'/config/'));
+                $loader->load('services.yml');
+            }
+        }
+    }
+
+    /**
+     * Allow the PrestaShop modules to override Twig templates
+     *
+     * @param ContainerBuilder $container
+     * @return void
+     */
+    private function registerTemplatesFromModules(ContainerBuilder $container)
+    {
+        $twigFilesystemLoaderDefinition = $container->getDefinition('twig.loader.filesystem');
+
+        $bundles = $this->getBundlesList($container);
+
+        foreach ($bundles as $bundle) {
+            $namespace = $this->normalizeBundleName($bundle);
+            $installedModules = $container->get('prestashop.core.admin.module.repository')->getInstalledModules();
```

Comment:
Not the best idea, but the only working solution I have regarding the multiple dependencies of `ModuleRepository`.

Reply:
so you are calling a service inside compilerpass where services are built. really crazy. ;)
as i see you only need the Twig namespaces here. abouts about a custom dynamic Twig template loader / locator?

Reply:
It's a really good idea! Thanks :-)

And yes, this is why I'm not so proud of this code :/

Reply:
Hmm, in fact I also need installed modules to limit the services loaded :/ not so easy to do :/